### PR TITLE
fix Issue 14656 - "auto" of "auto ref" spills over to other function

### PR DESCRIPTION
**Pull request by Kozzi11**
https://issues.dlang.org/show_bug.cgi?id=14656

### CODE REVIEW

**Comment by MetaLang** [comment]
14656 is blocking a pull request I'm working on, so thanks a lot for this.

File: test/runnable/functype.d [hunk 1]
```diff
@@ -296,6 +296,26 @@ void test10734()
 
 }
 
+void test14656()
+{
+    void unaryFun()(auto int a) pure nothrow @safe @nogc {}
```
Comment (9rnsr) [inline test/runnable/functype.d]: Currently applying only `auto` attribute on template function parameters does not make any errors, but I think it's incorrect.

This line should raise `Error: auto can only be used for template function parameters` as same as non-template functions.

Comment (9rnsr) [inline test/runnable/functype.d]: However, it would be a separated issue. This bugfix itself looks good to me.

**Comment by 9rnsr** [comment]
Auto-merge toggled on

**Pull request merged**